Feature/#48 asset minification

Exulansis opened the pull request:
I added the gulp-uglify package (https://goo.gl/dH5Zlz) so that the final size of the "app.js" file can be reduced. Modified the gulpfile.js file so that the the final "app.js" file is minified.

reederz commented:
@Exulansis nice work. However, we only want to minify the assets in production. Otherwise, the feedback loop for development will be way too long. Could you make a separate gulp task for this. It should be run like this:
```bash
gulp build-prod
```

Also, please add a note in `README.md' on how to use it.

Exulansis commented:
@reederz Thank you for the feedback. I modified the gulpfile accordly. 
I am not entirely sure if accessing the "app.js" file in the 'build-prod' task using the: `destinations.js + '/app.js'` line is very elegant / appropriate. 
The `gulp build-prod` task takes around 30 seconds on my machine so I now see how it can slow the feedback loop noticeably.

Do I add the node into the root 'README.md' file?

reederz commented:
@Exulansis Yeah, the naming thing is OK. One more thing: take a look at this task `gulp build`: https://github.com/jolocom/little-sister/blob/feature/%2348-asset-minification/gulpfile.js#L140

Right now, your `gulp build-prod` will only work, if one runs `gulp build` before. Can you make `gulp build-prod` identical to `gulp build` but with the addition of minification?

Re: `README.md` - yes, the root `README.md` is the place. Right now we have fairly little documentation (I know, it's bad) so the single `README.md` file is fine. But I suppose that later we'll move the docs to the github wiki.

Exulansis commented:
@reederz I added the 'build-prod' gulp task. Hope that looks better now. The 'default' task still executes the standard 'build' gulp task, so the 'build-prod' has to be ran using the `bash gulp build-prod` command. </br>As a result of that it is ran without the watcher. If we want to run that and have the watch task on it we have to modify the default gulp task to run 'build-prod' instead of 'build'. Not sure if that's a problem or not. </br> I also added a small passage in the `README.me` file regarding the new task. 
Feedback is always greatly appreciated!

reederz commented:
@Exulansis this is excellent! Our app is now much quicker to load.
